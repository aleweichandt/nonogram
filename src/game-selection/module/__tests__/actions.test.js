// @flow
import {
  chooseGame,
  choosePack,
  loadGames,
  loadPacks,
  storeGames,
  storePacks,
} from '../actions';

const testGame = {
  id: 'gameId',
  title: 'title',
  thumbnailUrl: 't-url',
  boardData: 'data',
};
const testPack = {
  id: 'packId',
  games: ['gameId'],
  title: 'title',
  description: 'desc',
  thumbnailUrl: 't-url',
  backgroundUrl: 'b-url',
};

describe('game selection actions test suite', () => {
  it('load packs', () => {
    expect(loadPacks()).toMatchSnapshot();
  });
  describe('store packs', () => {
    it('empty', () => {
      expect(storePacks()).toMatchSnapshot();
    });
    it('filled', () => {
      expect(storePacks([testPack])).toMatchSnapshot();
    });
  });
  it('choose pack', () => {
    expect(choosePack('packid')).toMatchSnapshot();
  });
  it('load pack games', () => {
    expect(loadGames('packid')).toMatchSnapshot();
  });
  describe('store pack games', () => {
    it('empty', () => {
      expect(storeGames()).toMatchSnapshot();
    });
    it('filled', () => {
      expect(storeGames([testGame])).toMatchSnapshot();
    });
  });
  it('choose game', () => {
    expect(chooseGame('gameid')).toMatchSnapshot();
  });
});
