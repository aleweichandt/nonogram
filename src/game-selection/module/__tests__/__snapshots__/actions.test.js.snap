// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`game selection actions test suite choose game 1`] = `
Object {
  "payload": Object {
    "gameId": "gameid",
  },
  "type": "game-selection/CHOOSE_GAME",
}
`;

exports[`game selection actions test suite choose pack 1`] = `
Object {
  "payload": Object {
    "packId": "packid",
  },
  "type": "game-selection/CHOOSE_PACK",
}
`;

exports[`game selection actions test suite load pack games 1`] = `
Object {
  "payload": Object {
    "packId": "packid",
  },
  "type": "game-selection/LOAD_GAMES",
}
`;

exports[`game selection actions test suite load packs 1`] = `
Object {
  "type": "game-selection/LOAD_PACKS",
}
`;

exports[`game selection actions test suite store pack games empty 1`] = `
Object {
  "payload": Object {
    "games": Array [],
  },
  "type": "game-selection/STORE_GAMES",
}
`;

exports[`game selection actions test suite store pack games filled 1`] = `
Object {
  "payload": Object {
    "games": Array [
      Object {
        "boardData": "data",
        "id": "gameId",
        "thumbnail": "t-url",
        "title": "title",
      },
    ],
  },
  "type": "game-selection/STORE_GAMES",
}
`;

exports[`game selection actions test suite store packs empty 1`] = `
Object {
  "payload": Object {
    "packs": Array [],
  },
  "type": "game-selection/STORE_PACKS",
}
`;

exports[`game selection actions test suite store packs filled 1`] = `
Object {
  "payload": Object {
    "packs": Array [
      Object {
        "backgroundUrl": "b-url",
        "description": "desc",
        "games": Array [
          "gameId",
        ],
        "id": "packId",
        "thumbnailUrl": "t-url",
        "title": "title",
      },
    ],
  },
  "type": "game-selection/STORE_PACKS",
}
`;
